text/0095-cognito-construct-library/working-backwards-readme.md

```diff
@@ -0,0 +1,567 @@
+>   The role assumption will be conditioned on a strict equals on an ExternalId that will be unique to this user pool.
+
+Code sample:
+
+```ts
+new UserPool(this, 'myuserpool', {
+  // ...
+  // ...
+  security: {
+    mfa: {
+      enforcement: MfaEnforcement.REQUIRED,
+      type: [ MfaType.SMS, MfaType.SOFTWARE_TOKEN ]
+    },
+    passwordPolicy: {
+      required: [ PasswordPolicy.LOWERCASE, PasswordPolicy.NUMBERS, PasswordPolicy.SYMBOLS ],
+      minLength: 12,
+    },
+    smsRole: iam.Role.fromRoleArn(/* ... */),
+  }
+});
+```
+
+> Internal Note: Password policy via UserPool-Policies, MFA enable via UserPool-MfaConfiguration; MFA type via
+> UserPool-EnabledMfas; smsRole via UserPool-SmsConfiguration
+
+> Internal Note: Account Recovery settings are missing from UserPool CloudFormation resource.
+
+### Emails
+
+Cognito will handle sending emails to users in the user pool. The address from which emails are sent can be configured
+on the user pool via the `from` property, and the `replyTo` property to configure the email where replies are sent.
+
+User pools can also be configured to send emails through Amazon SES, however, that is not yet supported via the CDK. Use
+the cfn layer to configure this - https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html.
+
+> These are the defaults that will be part of tsdoc, and not part of the README -
+> * email.from - TODO: find the default that Cognito uses.
+> * email.replyTo - same address as from.
+
+Code sample:
+
+```ts
+new UserPool(this, 'myuserpool', {
+  // ...
+  // ...
+  email: {
+    from: 'noreply@my-awesome-app.com'
+    replyTo: 'support@my-awesome-app.com'
+  }
+});
+```
+
+> Internal Note: configured via UserPool-EmailConfiguration
+
+### Triggers
+
+User pools can be configured with a number of lambda function backed triggers. They are available in the CDK via the
+`triggers` property. [Go
+here](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html)
+to read more about user pool workflows using lambda triggers, and details around each trigger.
+
+Check out the [documentation](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-cognito/lib/user-pool.ts#L148)
+to find the list of triggers supported by the CDK.
+
+Code sample:
+
+```ts
+const presignupLambda = new lambda.Function(this, 'presignup' {
+  // ...
+});
+
+const postsignupLambda = new lambda.Function(this, 'postsignup', {
+  // ...
+});
+
+new UserPool(this, 'myuserpool', {
+  // ...
+  // ...
+  triggers: {
+    preSignUp: presignupLambda,
+    postSignUp: postsignupLambda,
+    preAuthentication: preauthLambda,
+    // ...
+  }
+});
+```
+
+Triggers can also be added to the user pool outside of its constructor. This can be done via individual methods such as
+`addPreSgnUpTrigger()`, `addPostSignUpTrigger()`, `addPreAuthentication()`, etc.
+
```

Comment:
Maybe explain that the CDK adds permissions to the Lambdas so they can invoked by Cognito?

Reply:
Yes, good call out.